source/blood/src/nnextcitem.cpp

```diff
@@ -482,21 +482,14 @@ int userItemGetRespawnTime(spritetype* pSpr)
     return nTime;
 }
 
-char userItemViewUseRespawnMarkers(spritetype* pSpr)
+char userItemViewUseRespawnMarkers(ITEM* pItem)
 {
-    ITEM* pItem;
-    if ((pItem = userItemGet(pSpr->type)) != NULL)
+    if (pItem)
     {
         if (pItem->group == kItemGroupWeapon)
-        {
-            if (gGameOptions.nWeaponSettings == 3)
-                return 1;
-
-            return 0;
-        }
+            return (gGameOptions.nWeaponSettings == 3);
 
-        if (gGameOptions.nItemSettings == 2)
-            return 1;
+        return (gGameOptions.nItemSettings == 2);
```

Comment:
Good call on simplifying these return statements. It would be even better to remove the parentheses, because if someone made a typo and changed the `==` to `=`, having the statement wrapped in parentheses would silence compiler warnings about it.